packages/web/src/components/search/DataSearch.js

```diff
@@ -333,25 +334,7 @@ class DataSearch extends Component {
 		];
 	};
 
-	onSuggestions = (results) => {
-		const { parseSuggestion } = this.props;
-
-		const fields = Array.isArray(this.props.dataField)
-			? this.props.dataField
-			: [this.props.dataField];
-
-		const parsedSuggestions = getSuggestions(
-			fields,
-			results,
-			this.state.currentValue.toLowerCase(),
-		);
-
-		if (parseSuggestion) {
-			return parsedSuggestions.map(suggestion => parseSuggestion(suggestion));
-		}
-
-		return parsedSuggestions;
-	};
+	onSuggestions = results => handleOnSuggestions(results, this.state.currentValue, this.props);
```

Comment:
@ShahAnuj2610 This is not what we talked about, avoid passing `this.props` in functions.
Don't change the `parseSuggestion` logic here, just do the sorting part for promoted results in `getSuggestions` method itself.
We don't need any new method here.